src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -0,0 +1,128 @@
+            {
+                parentHandle = MetadataHelper.AddTypeReference(metadata, type,
+                    GetAssemblyReference(type.Assembly, metadata));
+            }
+
+            return parentHandle;
+        }
+
+        private AssemblyReferenceHandle GetAssemblyReference(Assembly assembly, MetadataBuilder metadata)
+        {
+            if (_assemblyRefStore.TryGetValue(assembly, out var handle))
+            {
+                return handle;
+            }
+
+            return MetadataHelper.AddAssemblyReference(assembly, metadata);
+        }
+        [RequiresAssemblyFiles("Returns <Unknown> for modules with no file path")]
+        public override string Name => _name;
+        public override string ScopeName => _name;
+        public override bool IsDefined(Type attributeType, bool inherit) => throw new NotImplementedException();
+        public override int GetFieldMetadataToken(FieldInfo field) => throw new NotImplementedException();
+        public override int GetMethodMetadataToken(ConstructorInfo constructor) => throw new NotImplementedException();
+        public override int GetMethodMetadataToken(MethodInfo method) => throw new NotImplementedException();
+        public override int GetStringMetadataToken(string stringConstant) => throw new NotImplementedException();
+        public override int GetTypeMetadataToken(Type type) => throw new NotImplementedException();
+        protected override void CreateGlobalFunctionsCore() => throw new NotImplementedException();
+        protected override EnumBuilder DefineEnumCore(string name, TypeAttributes visibility, Type underlyingType) => throw new NotImplementedException();
+        protected override MethodBuilder DefineGlobalMethodCore(string name, MethodAttributes attributes, CallingConventions callingConvention, Type? returnType, Type[]? requiredReturnTypeCustomModifiers, Type[]? optionalReturnTypeCustomModifiers, Type[]? parameterTypes, Type[][]? requiredParameterTypeCustomModifiers, Type[][]? optionalParameterTypeCustomModifiers) => throw new NotImplementedException();
+        protected override FieldBuilder DefineInitializedDataCore(string name, byte[] data, FieldAttributes attributes) => throw new NotImplementedException();
+        [RequiresUnreferencedCode("P/Invoke marshalling may dynamically access members that could be trimmed.")]
+        protected override MethodBuilder DefinePInvokeMethodCore(string name, string dllName, string entryName, MethodAttributes attributes, CallingConventions callingConvention, Type? returnType, Type[]? parameterTypes, CallingConvention nativeCallConv, CharSet nativeCharSet) => throw new NotImplementedException();
+        protected override TypeBuilder DefineTypeCore(string name, TypeAttributes attr, [DynamicallyAccessedMembers((DynamicallyAccessedMemberTypes)(-1))] Type? parent, Type[]? interfaces, PackingSize packingSize, int typesize)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(name);
+
```

Comment:
```suggestion
            ArgumentException.ThrowIfNullOrEmpty(name);

```
Argument validation should be taken care of by the non-core method. (It seems to be missing there now.)

Reply:
Yes, it is not checked there, do you want to update that with this PR?

Reply:
Sounds good to me.